### DISCUSSION TITLE
What is expected behavior for the "Portfolio Evolution" chart in the analysis section?

### DISCUSSION THREAD

**Discussion by shiruken** [ghostfolio/ghostfolio#1397] [Q&A]
Consider a hypothetical portfolio that consists of three transactions where a security is purchased and liquidated at a later date, with the proceeds used to purchase another security.

![Screenshot of Activities](https://user-images.githubusercontent.com/867617/197570765-64c02620-3116-48ae-b7b3-6e7dc0e226e5.png)

This is the resulting "Portfolio Evolution" chart shown on the portfolio analysis page:

![Screenshot of Portfolio Evolution Chart](https://user-images.githubusercontent.com/867617/197573916-be8d79fa-a206-4c1e-9ac6-8a24d864f051.png)

Is this expected behavior for this visualization? It seems like the SELL action is not being taken into account when calculating the account value. There is also no option for indicating _where_ the proceeds of the sale goes, which results in reinvestment appearing as a new deposit in the account.

**Comment by dtslvr** [comment]
Hey @shiruken
 
The chart indeed looks strange.

Would you also expect something like this for these activities?

<img width="1151" alt="portfolio-evolution" src="https://user-images.githubusercontent.com/4159106/197598664-26b62ae8-1a13-46ec-b00c-dc7ab9366492.png">

Exported activities: [ghostfolio-export-202210242027.json.zip](https://github.com/ghostfolio/ghostfolio/files/9854305/ghostfolio-export-202210242027.json.zip)

**Reply by shiruken** [reply]
Yes, I think that makes more sense although the wild swings in account value and deposits might be confusing, especially if money is withdrawn or not immediately reinvested. Would it be possible to maintain a "cash" balance in each account where the proceeds from SELL and DIVIDEND actions are held? We could then optionally pull from this balance during BUY actions instead of always being tracked as a deposit.

**Reply by dtslvr** [reply]
Test case: #1399

**Reply by dtslvr** [reply]
> Would it be possible to maintain a "cash" balance in each account where the proceeds from SELL and DIVIDEND actions are held?

This would probably be the most elegant solution based on double-entry bookkeeping. However, the tracking would also be more complicated. At the moment, you can just track the cash balance per account.

**Reply by shiruken** [reply]
Agreed. Definitely would be a "nice-to-have" but understandable if it's beyond the current scope of this project.

Does the cash balance per account track over time?

**Reply by dtslvr** [reply]
> Does the cash balance per account track over time?

No, it's currently just a single value.

By the way, the portfolio evolution chart has been improved in Ghostfolio [1.208.0](https://github.com/ghostfolio/ghostfolio/releases/tag/1.208.0).

**Reply by shiruken** [reply]
Looks good to me!

**Comment by francocalvo** [comment]
Hey! Since this isn't closed, I wanted to do a ping. I had a similar problem, and as a way to represent the cash balance, I "buy" USDT in the account. It's not fancy, but maybe treating USD as an asset would allow for it to work. For now I can't pull from that, I have to sell the USDT and buy my desired stock/crypto, but it works.
